src/ace.jsx

```diff
@@ -1,152 +1,105 @@
-var ace = require('brace');
-var React = require('react');
-
-module.exports = React.createClass({
-  displayName: 'ReactAce',
-
-  propTypes: {
-    mode: React.PropTypes.string,
-    theme: React.PropTypes.string,
-    name: React.PropTypes.string,
-    className: React.PropTypes.string,
-    height: React.PropTypes.string,
-    width: React.PropTypes.string,
-    fontSize: React.PropTypes.number,
-    showGutter: React.PropTypes.bool,
-    onChange: React.PropTypes.func,
-    onCopy: React.PropTypes.func,
-    onPaste: React.PropTypes.func,
-    onFocus: React.PropTypes.func,
-    onBlur: React.PropTypes.func,
-    value: React.PropTypes.string,
-    onLoad: React.PropTypes.func,
-    onBeforeLoad: React.PropTypes.func,
-    maxLines: React.PropTypes.number,
-    readOnly: React.PropTypes.bool,
-    highlightActiveLine: React.PropTypes.bool,
-    tabSize: React.PropTypes.number,
-    showPrintMargin: React.PropTypes.bool,
-    cursorStart: React.PropTypes.number,
-    editorProps: React.PropTypes.object,
-    keyboardHandler: React.PropTypes.string,
-    wrapEnabled: React.PropTypes.bool
-  },
-  getDefaultProps: function() {
-    return {
-      name: 'brace-editor',
-      mode: '',
-      theme: '',
-      height: '500px',
-      width: '500px',
-      value: '',
-      fontSize: 12,
-      showGutter: true,
-      onChange: null,
-      onPaste: null,
-      onLoad: null,
-      maxLines: null,
-      readOnly: false,
-      highlightActiveLine: true,
-      showPrintMargin: true,
-      tabSize: 4,
-      cursorStart: 1,
-      editorProps: {},
-      wrapEnabled:false
-    };
-  },
-  onChange: function() {
-    if (this.props.onChange && !this.silent) {
-      var value = this.editor.getValue();
-      this.props.onChange(value);
-    }
-  },
-  onFocus: function() {
-    if (this.props.onFocus) {
-      this.props.onFocus();
-    }
-  },
-  onBlur: function() {
-    if (this.props.onBlur) {
-      this.props.onBlur();
-    }
-  },
-  onCopy: function(text) {
-    if (this.props.onCopy) {
-      this.props.onCopy(text);
-    }
-  },
-  onPaste: function(text) {
-    if (this.props.onPaste) {
-      this.props.onPaste(text);
-    }
-  },
-  componentDidMount: function() {
-    this.editor = ace.edit(this.props.name);
-    if (this.props.onBeforeLoad) {
-      this.props.onBeforeLoad(ace);
+import ace from 'brace';
+import React, { Component, PropTypes } from 'react';
+
+export default class ReactAce extends Component {
+  constructor(props) {
+    super(props);
+    [
```

Comment:
ES6 class will not auto bind this for us, so we should bind this at constructor or for every call site.